### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by AnnikaStachowiak**
Pairing with @DaisyPalencia
Assignment 2 initial submission.

### CODE REVIEW

File: Logger/LogFactory.cs [hunk 1]
```diff
@@ -2,10 +2,25 @@
...
         {
-            
-            return null;
+            if (_pathName == null)
+            {
+                return null;
+            }
+            else
+            {
+                ConfigureFileLogger(_pathName);
```
Comment (jacobnewcomb) [inline Logger/LogFactory.cs:15]: This line is redundant.

**Review by jacobnewcomb — COMMENTED** [review]
- There is an existing BaseLogger class. It needs an auto property to hold a class name. This property should be set in the LogFactory using an object initializer. ✔
- Create a FileLogger that derives from BaseLogger. It should take in a path to a file to write the log message to. When its Log method is called, it should append messages on their own line in the file. The output should include all of the following:
        The current date/time ✔
        The name of the class that created the logger ✔
        The log level ✔
        The message ✔
- The LogFactory should be updated with a new method ConfigureFileLogger. This should take in a file path and store it in a private member. It should use this when instantiating a new FileLogger in its CreateLogger method. ✔
- If the file logger has not be configured in the LogFactory, its CreateLogger method should return null. ✔
- Inside of BaseLoggerMixins implement extension methods on BaseLogger for
        Error, ✔
        Warning, ✔
        Information, and ✔
        Debug. ✔
- Use the nameof() operator when identifying the class name to the logger ❌
- Ensure you turn on Warnings as Errors (TreatWarningsAsErrors) ✔
- Ensure that you turn on code analysis (EnableNETAnalyzers) ✔
- Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ✔
- Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
- Turn on Nullability (Nullable) ✔
- All of the above should be unit tested ✔

Would've liked to see you try extra credit, but I thought mostly all of the changes made are good! Just make sure to use nameof() when referring to the name of a class.

**Comment by BenagainB** [comment]
There is an existing BaseLogger class. It needs an auto property to hold a class name. This property should be set in the LogFactory using an object initializer. ✔
Create a FileLogger that derives from BaseLogger. It should take in a path to a file to write the log message to. When its Log method is called, it should append messages on their own line in the file. The output should include all of the following:
The current date/time ✔
The name of the class that created the logger ✔
The log level ✔
The message ✔
The LogFactory should be updated with a new method ConfigureFileLogger. This should take in a file path and store it in a private member. It should use this when instantiating a new FileLogger in its CreateLogger method. ✔
If the file logger has not be configured in the LogFactory, its CreateLogger method should return null. ✔
Inside of BaseLoggerMixins implement extension methods on BaseLogger for
Error, ✔
Warning, ✔
Information, and ✔
Debug. ✔
Use the nameof() operator when identifying the class name to the logger ✔
Ensure you turn on Warnings as Errors (TreatWarningsAsErrors) ✔
Ensure that you turn on code analysis (EnableNETAnalyzers) ✔
Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ✔
Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
Turn on Nullability (Nullable) ✔
All of the above should be unit tested ✔

**Comment by BenagainB** [comment]
Your classes are very straightforward and easy to understand what's going on. I think you're well ahead of me on unit tests and I'll probably revise mine after seeing how you implemented yours. Good job!

File: Logger/LogFactory.cs (hunk 1 above)
Comment (DaisyPalencia) [reply Logger/LogFactory.cs:15]: Thank you, we will be fixing the redundant line and adding nameof().

**Comment by DaisyPalencia** [comment]
> Your classes are very straightforward and easy to understand what's going on. I think you're well ahead of me on unit tests and I'll probably revise mine after seeing how you implemented yours. Good job!

Thank you!

File: Logger/LogFactory.cs [hunk 2]
```diff
@@ -2,10 +2,24 @@
...
     public class LogFactory
     {
-        public BaseLogger CreateLogger(string className)
+        private string? _pathName;
+
+        public BaseLogger? CreateLogger(string className)
         {
-            
-            return null;
+            if (_pathName == null)
```
Comment (mmwoodfo) [inline Logger/LogFactory.cs:9]:
```suggestion
            if (_pathName is null)
```
Can use `is` here aswell

File: Logger/BaseLoggerMixins.cs [hunk 3]
```diff
@@ -1,7 +1,52 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
     public static class BaseLoggerMixins
     {
-        
+        public static void Error(this BaseLogger logger, string message, params object[] paths)
```
Comment (mmwoodfo) [inline Logger/BaseLoggerMixins.cs:7]:
```suggestion
        public static void Error(this BaseLogger logger, string message, params object[] args)
```

**Pull request closed without merging**